src/jsonrpccpp/client/connectors/tcpsocketclientprivate.h

```diff
@@ -0,0 +1,26 @@
+/*************************************************************************
+ * libjson-rpc-cpp
+ *************************************************************************
+ * @file    tcpsocketclientprivate.h
+ * @date    17.07.2015
+ * @author  Alexandre Poirot <alexandre.poirot@legrand.fr>
+ * @license See attached LICENSE.txt
+ ************************************************************************/
+
+#ifndef JSONRPC_CPP_TCPSOCKETCLIENTPRIVATE_H_
+#define JSONRPC_CPP_TCPSOCKETCLIENTPRIVATE_H_
+
+#include "../iclientconnector.h"
+#include <jsonrpccpp/common/exception.h>
+
+namespace jsonrpc
+{
+        /**
+         * This class is an interface to the real implementation of TcpSocketClient. Kind of a strategy design pattern.
+         */
+	class TcpSocketClientPrivate : public IClientConnector
+	{
```

Comment:
Hi Peter !

I'm replying a bit late, I did not had the time to check out your comments.

I used this class (or let's say interface) just not to accept any IClientConnectors.

The "public" TcpSocketClient" uses a pointer to this class but it could be a pointer to a IClientConnector as well. The only thing that bugged me on using IClientConnector is that you could have modified the TcpSocketClient to use an instance of HttpClientConnector as pointer value.

It's kind of structural choice : TcpSocketClient uses this class as pointers and any connector that is a real implementation of tcp socket client should inherit it instead of IClientConnector, just to avoid confusion when reading real tcp socket implementation classe.